### PR TITLE
Upload images

### PR DESCRIPTION
**Pull request by Shahzzoda**
Uploading images is now integrated into the app using Cloudinary and is being pulled down on the Profile page.  Does not include the ability to order the photos, or tag location. No Map feature yet.

### CODE REVIEW

File: api/controllers/trips.js [hunk 1]
```diff
@@ -26,21 +26,24 @@ router.get('/:id', (req,res) => {
...
-
-    res.status(201).json({
-      trip,
-      user
-    });
-  } catch(e) {
-    res.status(400).json(e);
-  }
+router.post('/', (req, res) => {
+  let { content } = req.body;
```
Comment (jinlin314) [inline api/controllers/trips.js]: frontend should just send in {name, description, coverPhoto, pics} in flat object, not nested content object.

File: api/controllers/trips.js [hunk 2]
```diff
@@ -26,21 +26,24 @@ router.get('/:id', (req,res) => {
...
+  Trip.create({
+    name: content.name,
+    description: content.description,
+    coverphoto: content.coverPhoto,
+    userId: 1,
```
Comment (jinlin314) [inline api/controllers/trips.js]: do not send in userId in payload

File: api/controllers/trips.js [hunk 3]
```diff
@@ -26,21 +26,24 @@ router.get('/:id', (req,res) => {
...
+  })
+  .then((trip) => {
+    content.pics.forEach((el) => {
```
Comment (jinlin314) [inline api/controllers/trips.js]: please use Media.bulkCreate([picsArrayObject], {returning: true}) to insert all pics at once.

File: api/controllers/trips.js [hunk 4]
```diff
@@ -26,21 +26,24 @@ router.get('/:id', (req,res) => {
...
+      Media.create({name:'1233', description:'', photo: el})
```
Comment (jinlin314) [inline api/controllers/trips.js]: should send the name and description of the photos from frontend

File: api/controllers/trips.js [hunk 5]
```diff
@@ -26,21 +26,24 @@ router.get('/:id', (req,res) => {
...
+      .then((media) =>{
+        media.setTrip(trip);
```
Comment (jinlin314) [inline api/controllers/trips.js]: after bulkCreate, user Trip.addMedias(picsCreated)

File: client/src/components/ImageCard.js [hunk 6]
```diff
@@ -1,12 +1,12 @@
 import React from 'react';
-import sally from '../assets/images/japan.jpg';
 
 function ImageCard(props) {
   // import photo from String(".." + props.coverPhoto);
   // console.log(photo)
   return (
     <div className="col-4">
-        <img className="img-thumbnail img-responsive" src={sally} alt="sally's icon"/>
+    {console.log(props.src)}
```
Comment (jinlin314) [inline client/src/components/ImageCard.js]: remove console

File: client/src/pages/TripFormPage.js [hunk 7]
```diff
@@ -54,7 +60,43 @@ class TripFormPage extends React.Component {
       });
   }
 
+  picsAdded = (event) => {
```
Comment (jinlin314) [inline client/src/pages/TripFormPage.js]: you don't need this function

File: client/src/pages/TripFormPage.js [hunk 8]
```diff
@@ -31,7 +36,8 @@ class TripFormPage extends React.Component {
       body: JSON.stringify({content: {
```
Comment (jinlin314) [inline client/src/pages/TripFormPage.js]: rename this function to onPostTrip or onCreateTrip. 
do not stringify this.
call the create trip api, send in flat object of trip and pics info to create trip

File: client/src/pages/TripFormPage.js [hunk 9]
```diff
@@ -54,7 +60,43 @@ class TripFormPage extends React.Component {
       });
   }
 
+  picsAdded = (event) => {
+    this.state.pics.forEach(function(i){
+      console.log(i)
+    })
+  }
+
+  // componentDidMount() {
```
Comment (jinlin314) [inline client/src/pages/TripFormPage.js]: you can pass trip id in as props. If id is undefined, then is a create, if id is presented, then is update.  You can get it id from react router params in the page level, and pass it down to the form.
EX: const { match: { params } } = this.props;
cont id = params.id

File: client/src/pages/TripFormPage.js [hunk 10]
```diff
@@ -54,7 +60,43 @@ class TripFormPage extends React.Component {
...
+  //   const id = 1;
+  //   fetch("/api/trips/" + id)
```
Comment (jinlin314) [inline client/src/pages/TripFormPage.js]: on componentDidMount, if is update, call api the fetch the existing trip data in database and pre-populate the form

File: client/src/pages/TripFormPage.js [hunk 11]
```diff
@@ -84,8 +125,11 @@ class TripFormPage extends React.Component {
             className="form-control mr-3 rounded"
             onChange={this.nameChanged}
           />
-          <button className="btn btn-secondary" >Upload</button>
+          <button className="btn cloudinary-button" onClick={() => myWidget.open()}>Upload</button>
           <button className="btn btn-primary" onClick={this.savePost}>Post</button>
+          <div className="form-list">
+            {this.state.pics}
+          </div>
         </div>
```
Comment (jinlin314) [inline client/src/pages/TripFormPage.js]: this form is not complete right? where is the post Trip button. on click post button, should call the onPostTrip function we discuss earlier

File: client/src/pages/TripFormPage.js [hunk 12]
```diff
@@ -84,8 +125,11 @@ class TripFormPage extends React.Component {
...
       </div>
     );
```
Comment (jinlin314) [inline client/src/pages/TripFormPage.js]: and you need to handle the updating a existing trip in both frontend and backend as well

**Comment by Shahzzoda** [comment]
will come back and review this

**Pull request merged**